kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/BaseConnectorIT.java

```diff
@@ -0,0 +1,483 @@
+      File downloadedFile = new File(destinationPath);
+      log.info("Saving file to : {}", destinationPath);
+      S3Client.getObject(new GetObjectRequest(bucketName, file.getKey()), downloadedFile);
+
+      List<JsonNode> downloadedFileContents = getFileContents(destinationPath);
+      if (!fileContentsMatchExpected(downloadedFileContents, expectedRowsPerFile, expectedRow)) {
+        return false;
+      }
+      downloadedFile.delete();
+    }
+    return true;
+  }
+
+  /**
+   * Check if the contents of a downloaded file match the expected row.
+   *
+   * @param fileContents        the file contents as a list of JsonNodes
+   * @param expectedRowsPerFile the number of rows expected in the file
+   * @param expectedRow         the expected values of each row
+   * @return whether the file contents match the expected row
+   */
+  private boolean fileContentsMatchExpected(List<JsonNode> fileContents, int expectedRowsPerFile,
+      Struct expectedRow) {
+    if (fileContents.size() != expectedRowsPerFile) {
+      log.error("Number of rows in file do not match the expected count, actual: {}, expected: {}",
+          fileContents.size(), expectedRowsPerFile);
+      return false;
+    }
+    for (JsonNode row : fileContents) {
+      if (!fileRowMatchesExpectedRow(row, expectedRow)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Compare the row in the file and its values to the expected row's values.
+   *
+   * @param fileRow     the row read from the file as a JsonNode
+   * @param expectedRow the expected contents of the row
+   * @return whether the file row matches the expected row
+   */
+  private boolean fileRowMatchesExpectedRow(JsonNode fileRow, Struct expectedRow) {
+    log.debug("Comparing rows: file: {}, expected: {}", fileRow, expectedRow);
+    // compare the field values
+    for (Field key : expectedRow.schema().fields()) {
+      String expectedValue = expectedRow.get(key).toString();
+      String rowValue = fileRow.get(key.name()).toString().replaceAll("^\"|\"$", "");
+      log.debug("Comparing values: {}, {}", expectedValue, rowValue);
+      if (!rowValue.equals(expectedValue)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Get the contents of a downloaded file from S3 based on its file extension.
+   *
+   * @param filePath the path of the file to read
+   * @return the rows of the file as JsonNodes
+   */
+  private List<JsonNode> getFileContents(String filePath) throws IOException {
+    if (filePath.endsWith(".avro")) {
+      return getContentsFromAvro(filePath);
+    } else if (filePath.endsWith(".parquet")) {
+      return getContentsFromParquet(filePath);
+    } else if (filePath.endsWith(".json")) {
+      return getContentsFromJson(filePath);
+    } else {
+      throw new InvalidFileTypeException(
+          String.format("Downloaded file has unsupported extension: %s", filePath)
+      );
+    }
```

Comment:
if we wanna be fancy we can use a map of extensions to Function<> and call that like here https://github.com/confluentinc/kafka-connect-object-store-source/blob/master/cloud-storage-source-common/src/test/java/io/confluent/connect/cloud/storage/source/integration/StorageSourceConnectorIT.java#L49-L54 and here https://github.com/confluentinc/kafka-connect-object-store-source/blob/master/cloud-storage-source-common/src/test/java/io/confluent/connect/cloud/storage/source/integration/StorageSourceConnectorIT.java#L320-L328

Reply:
If-else seems okay for simplicity, since we'd want the exception to be thrown if none match.

Reply:
it can just throw NPE which is fine for ITs

Reply:
also allows for extension of tests when we want to test compression which have different extensions